src/welcome/welcome.ts

```diff
@@ -35,13 +37,49 @@ export class WelcomePageProvider extends Disposable implements TextDocumentConte
         return this._context.globalState.get<boolean>(GAUGE_SUPPRESS_WELCOME);
     }
 
+    getLinuxDistribution(): string {
+        let dist = spawnSync('cat', ['/etc/issue']);
+        if (dist.error) {
+            return null;
+        }
+        return dist.stdout.toString();
+    }
+
+    getInstallCommandBasedOnOS(): any {
+        let installCommand: any = {};
+        switch (process.platform) {
+            case "win32":
+                installCommand.name = "Chocolatey";
+                installCommand.command = "choco install gauge";
+                return installCommand;
+            case "darwin":
+                installCommand.name = "Brew";
+                installCommand.command = "brew install gauge";
+                return installCommand;
+            default:
+                if (this.getLinuxDistribution().indexOf("Ubuntu") !== -1) {
+                    installCommand.name = "apt-get";
+                    installCommand.command = "sudo apt-get install gauge";
+                    return installCommand;
+                } else {
+                    installCommand.name = "dnf";
+                    installCommand.command = "sudo dnf install gauge";
+                    return installCommand;
+                }
+        }
+    }
+
     async provideTextDocumentContent(uri: Uri): Promise<string> {
         let rootPath = path.join('out', uri.path);
         let docPath = Uri.file(this._context.asAbsolutePath(path.join(rootPath, 'index.html')));
         const doc = await workspace.openTextDocument(docPath);
         let text =  doc.getText();
         let supress = this._context.globalState.get<Boolean>(GAUGE_SUPPRESS_WELCOME);
         return text
+            .replace(/{{installCommand}}/g, encodeURI('command:gauge.createAndSendText.terminal?' +
```

Comment:
There are too many chained replace. It now makes sense to extract a method and unit test it.